sympy/crypto/crypto.py

```diff
@@ -256,8 +256,43 @@ def encipher_substitution(pt, key, symbols="ABCDEFGHIJKLMNOPQRSTUVWXYZ"):
 #################### Vigenère cipher examples ########################
 ######################################################################
 
+def check_and_format(phrase, repeats=True, symbols="ABCDEFGHIJKLMNOPQRSTUVWXYZ"):
+    """
+    Returns a string formatted in a way as needed by Vigenère encryption.
+
+    Parameters
+    ==========
 
-def encipher_vigenere(pt, key, symbols="ABCDEFGHIJKLMNOPQRSTUVWXYZ"):
+    phrase:     string to be formatted
+    repeats:    boolean to allow or restrict repetition of characters
+                in the formatted phrase
+    symbols:    alphabet of characters allowed in phrase
+                (a ValueError is returned if phrase includes
+                a character that is not defined in symbols)
+
+    Examples
+    ========
+
+    >>> from sympy.crypto.crypto import check_and_format
+    >>> check_and_format('with repetitions')
+    'WITHREPETITIONS'
+    >>> check_and_format('without repetitions', False)
+    'WITHOUREPNS'
+
+    """
+    for x in phrase:
```

Comment:
I think I would keep this simple, not distinguishing between upper and lower case, not even removing whitespace:

```python
def foo(phrase, repeats, symbols):
  squashed = ''.join(''.join(phrase))  # accepts a string or list of strings
  missing = set(squashed) - set(symbols)
  if missing:
    raise ValueError('the following characters are missing from the symbols: %s' % missing)
  return squashed if repeats else ''.join(uniq(squashed))
```

vigenere would call it as `foo(phrase.upper().split(), repeats. string.uppercase)`:

```python
>>> foo('try this'.upper().split(), True, string.uppercase)
'TRYTHIS'
>>> foo('try this'.upper().split(), False, string.uppercase)
'TRYHIS'
```
Perhaps a vigenere helper named `general_vigenere` could be made that is the generalized vigenere with no assumptions about allowed or unallowed symbols and the `encipher_vigenere` would apply the textbook encryption (e.g. no whitespace, all caps, etc...).